### PR TITLE
fix: (Platform) add subtle mode in core and platform. Fix vertical align in platform link.

### PR DESCRIPTION
**Pull request by DeepakSap14**
#### Please provide a link to the associated issue.
#3612 

#### Please provide a brief summary of this pull request.
1. Added missing subtle link in Core link and platform link.
2. fix for vertical alignment of platform link

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: apps/docs/src/app/platform/component-docs/platform-link/platform-link-examples/platform-link-standard-example.component.html [hunk 1]
```diff
@@ -7,6 +7,16 @@
     text after link
 </p>
 <br />
+
+<div>
+    This is an Example of
+    <fdp-link [href]="'http://www.google.com'" [title]="'Extra info as tooltip text'" linkType="subtle">
```
Comment (Betrozov) [inline apps/docs/src/app/platform/component-docs/platform-link/platform-link-examples/platform-link-standard-example.component.html:13]: `<fdp-link [href]="'http://www.google.com'" [title]="'Extra info as tooltip text'" linkType="subtle">`
=>
`<fdp-link href="http://www.google.com" title="Extra info as tooltip text" linkType="subtle">`

File: apps/docs/src/app/platform/component-docs/platform-link/platform-link-examples/platform-link-truncated-example.component.html [hunk 2]
```diff
@@ -1,8 +1,7 @@
 <div>
     This is an Example of
     <p style="display: inline-block; width: 150px;">
-        <fdp-link [href]="'http://www.google.com'" title="this is a long text link. so getting truncated.">
-            <fd-icon [glyph]="'chain-link'"></fd-icon>
+        <fdp-link [href]="'http://www.google.com'" [truncate]="true" title="this is a long text link. so getting truncated.">
```
Comment (Betrozov) [inline apps/docs/src/app/platform/component-docs/platform-link/platform-link-examples/platform-link-truncated-example.component.html:4]: the same

**Review by Betrozov — CHANGES_REQUESTED** [review]

**Review by sKudum — APPROVED** [review]

File: apps/docs/src/app/platform/component-docs/platform-link/platform-link-examples/platform-link-standard-example.component.html (hunk 1 above)
Comment (DeepakSap14) [reply apps/docs/src/app/platform/component-docs/platform-link/platform-link-examples/platform-link-standard-example.component.html]: fixed

File: apps/docs/src/app/platform/component-docs/platform-link/platform-link-examples/platform-link-truncated-example.component.html (hunk 2 above)
Comment (DeepakSap14) [reply apps/docs/src/app/platform/component-docs/platform-link/platform-link-examples/platform-link-truncated-example.component.html]: fixed

**Review by KevinOkamoto — APPROVED** [review]
LGTM

File: libs/platform/src/lib/components/link/link.component.ts [hunk 3]
```diff
@@ -144,6 +150,8 @@ export class LinkComponent extends BaseComponent implements OnInit, AfterViewIni
         /* If link linkType===emphasized then make link emphasized type */
         if (this.linkType === VALID_INPUT_TYPES[1]) {
             this.emphasized = true;
+        } else if (this.linkType === VALID_INPUT_TYPES[2]) {
+            this.subtle = true;
         }
```
Comment (IlyaSurmay) [inline libs/platform/src/lib/components/link/link.component.ts:155]: looks like the whole thing could be replaced with just 2 lines
```
this.emphasized = this.linkType === VALID_INPUT_TYPES[1];
this.subtle = this.linkType === VALID_INPUT_TYPES[2];
```

**Review by IlyaSurmay — APPROVED** [review]
looks good

**Review by katekozlowska — APPROVED** [review]
LGTM

**Review by JKMarkowski — APPROVED** [review]

**Review by Betrozov — APPROVED** [review]

File: libs/platform/src/lib/components/link/link.component.ts (hunk 3 above)
Comment (DeepakSap14) [reply libs/platform/src/lib/components/link/link.component.ts]: fixed

**Comment by DeepakSap14** [comment]
have incorporated Icon component changes now.

**Pull request merged**